tapdb/sqlc/migrations/000015_tapscript_trees.up.sql

```diff
@@ -0,0 +1,28 @@
+CREATE TABLE IF NOT EXISTS tapscript_roots (
+        root_id BIGINT PRIMARY KEY,
+
+        root_hash BLOB NOT NULL UNIQUE CHECK(length(root_hash) = 32),
+
+        branch_only BOOLEAN NOT NULL DEFAULT FALSE
+);
+
+-- A script leaf can be referenced from multiple trees.
+CREATE TABLE IF NOT EXISTS tapscript_nodes (
+        node_id BIGINT PRIMARY KEY,
+
+        raw_node BLOB NOT NULL UNIQUE
+);
+
+CREATE TABLE IF NOT EXISTS tapscript_edges (
+        edge_id BIGINT PRIMARY KEY,
+
+        root_hash_id BIGINT NOT NULL REFERENCES tapscript_roots(root_id),
+
+        node_index BIGINT NOT NULL,
+
+        raw_node_id BIGINT NOT NULL REFERENCES tapscript_nodes(node_id),
+
+        -- A leaf can be repeated within a tree, and shared amongst trees, but
+        -- there can only be one leaf at a given index in a tree.
+        UNIQUE(root_hash_id, node_index, raw_node_id)
```

Comment:
I know this inline style for unique indexes is very nice. But the downside is that we cannot change such a unique constraint. So if we ever needed to add or remove a column here, we'd need to create a whole new table, copy over the data and so on.
So perhaps for new code we want to create a named index instead? Then that can be dropped and re-created much more easily.

Reply:
SGTM, you mean like:

https://github.com/lightninglabs/taproot-assets/blob/a6415f8e5cfafb1d7e85f09428905b80607c9be7/tapdb/sqlc/migrations/000014_multiverse_tree.up.sql#L38

?